### PR TITLE
[NCL-6415] Support RSQL ManyToMany filtering

### PR DESCRIPTION
**Pull request by dominikbrazdil**
### Checklist:

* [x] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java [hunk 1]
```diff
@@ -117,7 +117,6 @@ public void setPNCVersion(String version) {
         GenericSetting pncVersion = createGenericParameterIfNotFound(PNC_VERSION);
         pncVersion.setValue(version);
         genericSettingRepository.save(pncVersion);
-        notifier.sendMessage(GenericSettingNotification.newAnnoucement(version));
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java:120]: why removing this?

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java [hunk 2]
```diff
@@ -65,6 +68,13 @@ public AbstractRSQLMapper(Class<DB> type) {
                 return mapEntity(from, toEntity(name), selector.next());
             }
         }
+        if (toEntitySet(name) != null) {
+            if (selector.isFinal()) {
+                return (PluralAttributePath<?>) from.get(toEntitySet(name));
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java:73]: Why are you casting it to `PluralAttributePath` when the return type is only `Path`?

File: facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java (hunk 1 above)
Comment (dominikbrazdil) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java:120]: This was an error from my previous PR which a noticed radnomly now and did not want to create a special PR for it.

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java (hunk 2 above)
Comment (dominikbrazdil) [reply facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java:73]: I wanted it to be more specific to make it clear that its for multiple attributes, and from.get returns Expression so some cast is need. You think that casting to Path<> would be better?

File: facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java (hunk 1 above)
Comment (janinko) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java:120]: can you put it in separate commit at least then?

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java (hunk 2 above)
Comment (janinko) [reply facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/AbstractRSQLMapper.java:73]: yes, this is unnecessarily too specific.

File: facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java (hunk 1 above)
Comment (dominikbrazdil) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GenericSettingProvider.java:120]: Sure

**Review by janinko — APPROVED** [review]

**Pull request merged**